lib/Chunk.js

```diff
@@ -13,18 +13,52 @@ const ERR_CHUNK_ENTRY = "Chunk.entry was removed. Use hasRuntime()";
 const ERR_CHUNK_INITIAL =
 	"Chunk.initial was removed. Use canBeInitial/isOnlyInitial()";
 
+/** @typedef {import("./Module.js")} Module */
+/** @typedef {import("./ChunkGroup")} ChunkGroup */
+/** @typedef {import("./ModuleReason.js")} ModuleReason */
+/** @typedef {import("webpack-sources").Source} Source */
+
+/**
+ *  @typedef {Object} Identifiable an object who contains an identifier function property
+ *  @property {() => string} identifier the resource or unique identifier of something
+ */
+
+/**
+ *  @typedef {Object} WithId an object who has an id property
+ *  @property {string} id the id of the object
+ */
+
+/** @typedef {(a: Module, b: Module) => -1|0|1} ModuleSortPredicate */
+/** @typedef {(m: Module) => boolean} ModuleFilterPredicate */
+/** @typedef {(c: Chunk) => boolean} ChunkFilterPredicate */
+
+/**
+ * @param {WithId} a object that contains an ID property
+ * @param {WithId} b object that contains an ID property
+ * @returns {-1|0|1} sort value
+ */
 const sortById = (a, b) => {
 	if (a.id < b.id) return -1;
 	if (b.id < a.id) return 1;
 	return 0;
 };
 
+/**
+ *
+ * @param {Identifiable} a first object with ident fn
+ * @param {Identifiable} b second object with ident fn
+ * @returns {-1|0|1} The order number of the sort
+ */
 const sortByIdentifier = (a, b) => {
 	if (a.identifier() > b.identifier()) return 1;
 	if (a.identifier() < b.identifier()) return -1;
 	return 0;
 };
 
+/**
+ * @returns {string} a concatenation of module identifiers sorted
+ * @param {SortableSet} set to pull module identifiers from
```

Comment:
@mohsen1  Cannot find the `typedef` for `SortableSet`. Is it not needed here?

Reply:
It's imported on top of the file